server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/EnableAllocationDecider.java

```diff
@@ -147,17 +147,29 @@ public Decision canRebalance(RoutingAllocation allocation) {
             return allocation.decision(Decision.YES, NAME, "allocation is explicitly ignoring any disabling of rebalancing");
         }
 
-        if (enableRebalance == Rebalance.NONE) {
-            for (IndexMetaData indexMetaData : allocation.metaData()) {
-                if (INDEX_ROUTING_REBALANCE_ENABLE_SETTING.exists(indexMetaData.getSettings())
-                    && INDEX_ROUTING_REBALANCE_ENABLE_SETTING.get(indexMetaData.getSettings()) != Rebalance.NONE) {
-                    return allocation.decision(Decision.YES, NAME, "rebalancing is permitted on one or more indices");
+        Rebalance enable = this.enableRebalance;
+        switch (enable) {
+            case ALL:
+                return allocation.decision(Decision.YES, NAME, "all rebalance are allowed");
+            case NONE:
+                return allocation.decision(Decision.NO, NAME, "none rebalance are not allowed");
+            case PRIMARIES:
+                if (allocation.routingNodes().hasInactivePrimaries()) {
+                    return allocation.decision(Decision.NO, NAME,
```

Comment:
why does this globally disable rebalancing when there are some inactive primaries of an unrelated index?